modules/gapi/src/backends/onnx/gonnxbackend.cpp

```diff
@@ -893,7 +917,30 @@ void ONNXCompiled::Run(const std::vector<cv::Mat>& ins,
     GAPI_Assert(input_names.size() == this_session.GetInputCount());
 
     auto in_run_names  = getCharNames(input_names);
-    if (!is_dynamic && !is_postproc) {
+    if (out_tensor_info[0].type == ONNX_TENSOR_ELEMENT_DATA_TYPE_INT64) {
```

Comment:
Don't think we need it as a separate condition. Can we simply check as part of `else if (!is_dynamic && !is_postproc)` after `this_session.Run`? If `out_tensor_info[0].type == ONNX_TENSOR_ELEMENT_DATA_TYPE_INT64` then we know for sure we need to copy data back otherwise it's already in-place

Reply:
ok, I've changed this